swagger2.0/oic.r.rule.expression.swagger.json

```diff
@@ -0,0 +1,153 @@
+      "x-copyright": "copyright 2019 Open Connectivity Foundation, Inc. All rights reserved."
+    },
+    "termsOfService": "https://openconnectivityfoundation.github.io/core/DISCLAIMER.md"
+  },
+  "schemes": ["http"],
+  "consumes": ["application/json"],
+  "produces": ["application/json"],
+  "paths": {
+    "/RuleExpressionResURI": {
+      "get": {
+        "description": "Expression for the Rule that defines the Rule logic in terms of the defined Rule Inputs, and which evaluates to a boolean value, for which \"true\" means that the Rule has been triggered.",
+        "parameters": [
+          {"$ref": "#/parameters/interface-all"}
+        ],
+        "responses": {
+          "200": {
+            "description" : "Retrieves the Rule expression.",
+            "x-example":
+            {
+              "rt": ["oic.r.rule.expression"],
+              "if": ["oic.if.rw", "oic.if.baseline"],
+              "rule": "(timeperiod:triggered = true)",
+              "ruleresult": false,
+              "ruleenable": true,
+              "actionenable": true
+            },
+            "schema": { "$ref": "#/definitions/ruleexpression" }
+          }
+        }
+      },
+      "post": {
+        "description": "",
+        "parameters": [
+          {"$ref": "#/parameters/interface-rw"},
+          {
+            "name": "body",
+            "in": "body",
+            "required": true,
+            "schema": { "$ref": "#/definitions/ruleexpression-update" },
+            "x-example":
+              {
+                "ruleenable": true,
+                "actionenable": false
+              }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description" : "",
+            "x-example": {
+              "rule": "(timeperiod:triggered = true)",
+              "ruleresult": false,
+              "ruleenable": true,
+              "actionenable": false
+            },
+            "schema": { "$ref": "#/definitions/ruleexpression" }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "interface-rw": {
+      "in": "query",
+      "name": "if",
+      "type": "string",
+      "enum": ["oic.if.rw"]
+    },
+    "interface-all": {
+      "in": "query",
+      "name": "if",
+      "type": "string",
+      "enum": ["oic.if.rw","oic.if.baseline"]
+    }
+  },
+  "definitions": {
+    "ruleexpression" : {
+      "properties": {
+        "rt": {
+          "description": "The Resource Type.",
+          "items": {
+            "enum": ["oic.r.rule.expression"],
+            "maxLength": 64,
+            "type": "string"
+          },
+          "minItems": 1,
+          "uniqueItems": true,
+          "readOnly": true,
+          "type": "array"
+        },
+        "rule": {
+          "description": "The logical expression to be evaluated, see BNF",
+          "type": "string"
+        },
+        "ruleresult": {
+          "type": "boolean",
+          "readOnly": true
```

Comment:
Isn't this read-write?  It shows as RW in the CR at least.

Reply:
Probably ... I think we allow manual setting of this to test the rule.  Let me check.